File: 11-container-with-most-water.go
func min(l, r int) int {
    if l < r{
        return l
    } else {
        return r
    }
}

func maxArea(height []int) int {
    if len(height) < 2{
        return 0
    }
    left, right := 0, len(height) - 1
    max_area := min(height[left], height[right]) * (right - left)
    for left < right {
        area := min(height[left], height[right]) * (right - left)
        if area > max_area {
            max_area = area
        }
        if height[left] > height[right] {
            right -= 1
        } else {
            left += 1
        }
    }
    return max_area
}

File: bwu/math/11-container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        res = 0
        n = len(height)
        if n < 2:
            return res
            
        l, r = 0, n - 1
        while l < r:
            res = max(res, (r - l) * min(height[r], height[l]))
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
                
        return res
